Backend/app/kafka-consumers/src/Consumer/Flow.hs

```diff
@@ -105,6 +110,66 @@ availabilityConsumer flowRt appEnv kafkaConsumer =
         start = pure ([], Nothing)
         extract = pure
 
+becknRequestConsumer :: BRProcessor.BecknRequestType -> L.FlowRuntime -> AppEnv -> Consumer.KafkaConsumer -> IO ()
+becknRequestConsumer becknRequestType flowRt appEnv kafkaConsumer = do
+  flip finally (Consumer.closeConsumer kafkaConsumer) $ do
+    readBecknRequestMessages kafkaConsumer
+      >>= riderDrainerWithFlow
+      >>= commitAllTopics
+  where
+    riderDrainerWithFlow messages = do
+      runFlowR flowRt appEnv $
+        generateGUID
+          >>= flip withLogTag (BRProcessor.becknRequestProcessor becknRequestType messages)
+      pure messages
+
+    commitAllTopics messages = case messages of
+      [] -> print ("Nothing to commit" :: Text)
+      _ -> do
+        mbErr <- Consumer.commitAllOffsets Consumer.OffsetCommit kafkaConsumer
+        whenJust mbErr $ \err -> print ("Error while commit: message: " <> show err :: Text)
+
+readBecknRequestMessages ::
+  Consumer.KafkaConsumer ->
+  IO [BR.BecknRequestKafka]
+readBecknRequestMessages kafkaConsumer = do
+  pollStartTime <- getCurrentTime
+  let currentTopic = BR.countTopicNumber pollStartTime
+  eitherRecords <- pollMessageLoop (currentTopic, 0, [])
+  let records = mapMaybe hush eitherRecords
+  mbDecodedRecords <- forM records $ \record -> do
+    case Consumer.crValue record of
+      Just value -> do
+        case A.eitherDecode @BR.BecknRequestKafka . LBS.fromStrict $ value of
+          Right v -> pure (Just v)
+          Left err -> throwIO (InternalError $ "Could not decode record: " <> show value <> "; message: " <> show err) -- only when beckn_request tabular type was changed
+      Nothing -> pure Nothing
+  pure $ catMaybes mbDecodedRecords
+  where
+    pollMessageLoop :: (Int, Int, [Either Consumer.KafkaError ConsumerRecordD]) -> IO [Either Consumer.KafkaError ConsumerRecordD]
+    pollMessageLoop (currentTopic, emptyMessagesCount, messages) = do
+      eMessage <- Consumer.pollMessage kafkaConsumer (Consumer.Timeout 500)
+      (continueLoop, updEmptyMessagesCount) <- predicate currentTopic emptyMessagesCount eMessage
+      let updMessages = eMessage : messages
```

Comment:
and it's not streamly ? then looks like its going to use a lot of memory in prod environment.

Reply:
Fixed